arviz/kde_utils.py

```diff
@@ -0,0 +1,812 @@
+        It is the recommended approach when there are multiple modalities
+        with different spread.
+        It is not compatible with convolution. Defaults to False.
+    extend: boolean, optional
+        Whether to extend the observed range for `x` in the estimation.
+        It extends each bound by a multiple of the standard deviation of `x`
+        given by `extend_fct`. Defaults to False.
+    bound_correction: boolean, optional
+        Whether to perform boundary correction on the bounds of `x` or not.
+        Defaults to True.
+    extend_fct: float, optional
+        Number of standard deviations used to widen the
+        lower and upper bounds of `x`. Defaults to 0.5.
+    bw_fct: float, optional
+        A value that multiplies `bw` which enables tuning smoothness by hand.
+        Must be positive. Values below 1 decrease smoothness while values
+        above 1 decrease it. Defaults to 1 (no modification).
+    bw_return: bool, optional
+        Whether to return the estimated bandwidth in addition to the
+        other objects. Defaults to False.
+    custom_lims: list or tuple, optional
+        A list or tuple of length 2 indicating custom bounds
+        for the range of `x`. Defaults to None which disables custom bounds.
+    cumulative: bool, optional
+        Whether return the PDF or the cumulative PDF. Defaults to False.
+    grid_len: int, optional
+        The number of intervals used to bin the data points
+        (a.k.a. the length of the grid used in the estimation)
+        Defaults to 512.
+
+    Returns
+    -------
+    grid : Gridded numpy array for the x values.
+    pdf : Numpy array for the density estimates.
+    bw: optional, the estimated bandwidth.
+    """
+
+    # Check `x` is from appropiate type
+    x = check_type(x)
+
+    # Assert `bw_fct` is numeric and positive
+    assert isinstance(bw_fct, (int, float))
+    assert bw_fct > 0
+
+    # Preliminary calculations
+    x_len = len(x)
+    x_min = x.min()
+    x_max = x.max()
+    x_std = (((x ** 2).sum() / x_len) - (x.sum() / x_len) ** 2) ** 0.5
+    x_range = x_max - x_min
+
+    # Determine grid
+    grid_min, grid_max, grid_len = get_grid(
+        x_min, x_max, x_std, extend_fct, grid_len,
+        custom_lims, extend, bound_correction
+    )
+    grid_counts, _, grid_edges = histogram(x, grid_len, (grid_min, grid_max))
+
+    # Bandwidth estimation
+    bw = bw_fct * get_bw(x, bw, grid_counts, x_std, x_range)
+
+    # Density estimation
+    if adaptive:
+        grid, pdf = _kde_adaptive(x, bw, grid_edges, grid_counts, grid_len,
+                                  bound_correction)
+    else:
+        grid, pdf = _kde_convolution(x, bw, grid_edges, grid_counts, grid_len,
+                                     bound_correction)
+
+    if cumulative:
+        pdf = pdf.cumsum() / pdf.sum()
+
+    if bw_return:
+        return grid, pdf, bw
+    else:
+        return grid, pdf
+
+
+def kde_circular(
+    x,
+    bw="default",
```

Comment:
Could we have a different name? "circular scott", "taylor"...

Reply:
Since Scott's rule of thumb is named after the guy who proposed the Gaussian rule of thumb, I guess Taylor's rule of thumb is fine for this Von mises rule of thumb in the circular setting. Then `taylor` sounds good to me. Thanks for the suggestion.